agent/agent.go

```diff
@@ -567,7 +567,7 @@ func (a *Agent) Start(ctx context.Context) error {
 
 	// gRPC calls are only rate-limited on server, not client agents.
 	var grpcRateLimiter rpcRate.RequestLimitsHandler
-	grpcRateLimiter = rpcRate.NullRateLimiter()
+	grpcRateLimiter = rpcRate.NullRequestLimitsHandler()
```

Comment:
I made this change because I originally was going to create a noop Rate Limiter and the fact that this is named NullRateLimiter when it is based on RequestLimitsHandler interface and no RateLimiter interface was going to cause confusion.
Anyway, this is now more accurate, but is not required for this PR.

This PR is essentially [this commit](https://github.com/hashicorp/consul/pull/15819/commits/2c24f08bda96a9cce31ba3942b471ec29fe0e37b)